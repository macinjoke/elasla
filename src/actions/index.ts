import { Action, ActionCreator, AnyAction } from 'redux'
import { ThunkAction } from 'redux-thunk'
import { State } from '../reducers'

export const hogeAction = (): Action<string> => ({
  type: 'hogeAction',
})

export const login = (
  username: string,
  password: string,
): ThunkAction<void, State, undefined, Action<string>> => async dispatch => {
  const response = await fetch('http://localhost:3000/api/login', {
    method: 'post',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ username, password }),
  })
  if (response.status === 401) {
    console.log('パスワードが違います (本当はレンダリングしたい)')
  }
  if (response.status !== 200) {
    console.log(await response.text())
    return
  }
  const body = await response.json()
  localStorage.setItem('jwt', body.jwt)
  dispatch(loginSuccess(body))
}

export const loginSuccess = (body: object): AnyAction => ({
  type: 'loginSuccess',
  body,
})

export const fetchLoginState = (): ThunkAction<
  void,
  State,
  undefined,
  Action<string>
> => async dispatch => {
  const jwt = localStorage.getItem('jwt')
  if (!jwt) return

  const response = await fetch('http://localhost:3000/api/login', {
    headers: { authorization: `Bearer ${jwt}` },
  })
  if (response.status !== 200) {
    console.log(await response.json()) // TODO
  }
  const body = await response.json()
  dispatch(loginSuccess({ ...body, jwt }))
}

export const logout = (): ThunkAction<
  void,
  State,
  undefined,
  Action<string>
> => dispatch => {
  localStorage.removeItem('jwt')
  dispatch(logoutState())
}

export const logoutState = (): Action<string> => ({ type: 'logoutState' })
