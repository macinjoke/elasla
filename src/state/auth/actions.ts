import { Action as ReduxAction, AnyAction } from 'redux'
import { User } from './reducers'

export interface LoginSuccessAction {
  type: 'loginSuccess'
  body: User
}
export const loginSuccess = (body: any): LoginSuccessAction => ({
  type: 'loginSuccess',
  body,
})

export interface LoginFailureAction {
  type: 'loginFailure'
}
export const loginFailure = (): LoginFailureAction => ({ type: 'loginFailure' })

export interface LogoutStateAction {
  type: 'logoutState'
}
export const logoutState = (): LogoutStateAction => ({ type: 'logoutState' })

export type Action = LoginSuccessAction | LoginFailureAction | LogoutStateAction
