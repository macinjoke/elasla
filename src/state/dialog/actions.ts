import actionCreatorFactory from 'typescript-fsa'

const actionCreator = actionCreatorFactory('dialog')

export const openSignUpDialog = actionCreator('OPEN_SIGNUP_DIALOG')
export const closeSignUpDialog = actionCreator('CLOSE_SIGNUP_DIALOG')

export const openRegistrationFinishDialog = actionCreator(
  'OPEN_REGISTRATION_FINISH_DIALOG',
)
export const closeRegistrationFinishDialog = actionCreator(
  'CLOSE_REGISTRATION_FINISH_DIALOG',
)
