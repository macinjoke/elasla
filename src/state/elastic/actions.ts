import { Source } from './reducers'

export const SEARCH_SUCCESS = 'elastic/SEARCH_SUCCESS'
export interface SearchSuccessAction {
  type: typeof SEARCH_SUCCESS
  sources: Source[]
}
export const searchSuccess = (sources: Source[]): SearchSuccessAction => ({
  type: SEARCH_SUCCESS,
  sources,
})

export type Action = SearchSuccessAction
