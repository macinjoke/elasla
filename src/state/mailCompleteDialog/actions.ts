import actionCreatorFactory from 'typescript-fsa'

const actionCreator = actionCreatorFactory('mailCompleteDialog')

export const openDialog = actionCreator('OPEN__DIALOG')
export const closeDialog = actionCreator('CLOSE_DIALOG')
