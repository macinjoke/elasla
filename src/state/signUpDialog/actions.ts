import actionCreatorFactory from 'typescript-fsa'

const actionCreator = actionCreatorFactory('signUpDialog')

export const openDialog = actionCreator('OPEN_DIALOG')
export const closeDialog = actionCreator('CLOSE_DIALOG')
