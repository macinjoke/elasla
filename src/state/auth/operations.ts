import { Action } from 'redux'
import { State, ThunkAction } from '../../types'
import { loginFailure, loginSuccess, logoutState } from './actions'

export const login = (
  username: string,
  password: string,
): ThunkAction => async dispatch => {
  const response = await fetch('http://localhost:3000/api/login', {
    method: 'post',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ username, password }),
  })
  if (response.status === 401) {
    dispatch(loginFailure())
  }
  if (response.status !== 200) {
    console.log(await response.text())
    return
  }
  const body = await response.json()
  localStorage.setItem('jwt', body.jwt)
  dispatch(loginSuccess(body))
}

export const fetchLoginState = (): ThunkAction => async dispatch => {
  const jwt = localStorage.getItem('jwt')
  if (!jwt) return

  const response = await fetch('http://localhost:3000/api/login', {
    headers: { authorization: `Bearer ${jwt}` },
  })
  // localStorage.jwt が有効なものではなかった場合削除する
  if (response.status === 400 || response.status === 401) {
    localStorage.removeItem('jwt')
    return
  }
  if (response.status !== 200) return
  const body = await response.json()
  dispatch(loginSuccess({ ...body, jwt }))
}

export const logout = (): ThunkAction => dispatch => {
  localStorage.removeItem('jwt')
  dispatch(logoutState())
}
