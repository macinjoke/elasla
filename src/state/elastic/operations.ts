import { Client, SearchResponse } from 'elasticsearch'
import { Action } from 'redux'
import { CONFIG } from '../../constants'
import { ThunkAction } from '../../types'
import { searchSuccess } from './actions'
import { Source } from './reducers'

export const search = (): ThunkAction => async (dispatch, getState) => {
  const { host, port } = CONFIG.elasticsearch
  const client = new Client({
    host: `${host}:${port}`,
    log: 'trace',
  })
  const response: SearchResponse<Source> = await client.search({
    index: 'slack-*',
    type: 'slack-message',
    body: {
      query: {
        match: {
          text: getState().elastic.text,
        },
      },
    },
  })
  const sources = response.hits.hits.map((hit: any) => hit._source)
  dispatch(searchSuccess(sources))
}
