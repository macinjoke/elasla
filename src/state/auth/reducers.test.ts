;(global as any).localStorage = { getItem: () => 'some jwt text' }
import reducerTester from 'reducer-tester'
import { logout } from './actions'
import { fetchSession, login } from './operations'
import reducer, { initialState } from './reducers'

reducerTester({
  reducer,
  state: initialState,
  tests: [
    logout(),
    login.async.started({ username: 'foo', password: 'bar' }),
    login.async.failed({
      params: { username: 'foo', password: 'bar' },
      error: new Error('some error'),
    }),
    login.async.done({
      params: {
        username: 'foo',
        password: 'bar',
      },
      result: {
        username: 'foo',
        isMailAuthed: true,
        jwt: 'some jwt text',
      },
    }),
    fetchSession.async.started('some jwt text'),
    fetchSession.async.done({
      params: 'some jwt text',
      result: { username: 'foo', isMailAuthed: true, jwt: 'some jwt text' },
    }),
    fetchSession.async.failed({
      params: 'some jwt text',
      error: new Error('some error'),
    }),
  ],
})
