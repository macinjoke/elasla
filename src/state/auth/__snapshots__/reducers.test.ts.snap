// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth/FetchSession_DONE 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchSessionError": null,
    "isFetchingSession": false,
-   "isLogin": false,
+   "isLogin": true,
    "isLoginRequest": false,
    "loginError": null,
    "user": Object {
+     "isMailAuthed": true,
      "jwt": "some jwt text",
+     "username": "foo",
    },
  }
`;

exports[`auth/FetchSession_FAILED 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
-   "fetchSessionError": null,
+   "fetchSessionError": [Error: some error],
    "isFetchingSession": false,
    "isLogin": false,
    "isLoginRequest": false,
    "loginError": null,
    "user": Object {
-     "jwt": "some jwt text",
+     "jwt": "",
    },
  }
`;

exports[`auth/FetchSession_STARTED 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchSessionError": null,
-   "isFetchingSession": false,
+   "isFetchingSession": true,
    "isLogin": false,
    "isLoginRequest": false,
    "loginError": null,
    "user": Object {
      "jwt": "some jwt text",
    },
  }
`;

exports[`auth/LOGOUT 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchSessionError": null,
    "isFetchingSession": false,
    "isLogin": false,
    "isLoginRequest": false,
    "loginError": null,
-   "user": Object {
-     "jwt": "some jwt text",
-   },
+   "user": Object {},
  }
`;

exports[`auth/Login_DONE 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchSessionError": null,
    "isFetchingSession": false,
-   "isLogin": false,
+   "isLogin": true,
    "isLoginRequest": false,
    "loginError": null,
    "user": Object {
+     "isMailAuthed": true,
      "jwt": "some jwt text",
+     "username": "foo",
    },
  }
`;

exports[`auth/Login_FAILED 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchSessionError": null,
    "isFetchingSession": false,
    "isLogin": false,
    "isLoginRequest": false,
-   "loginError": null,
+   "loginError": [Error: some error],
    "user": Object {
      "jwt": "some jwt text",
    },
  }
`;

exports[`auth/Login_STARTED 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchSessionError": null,
    "isFetchingSession": false,
    "isLogin": false,
-   "isLoginRequest": false,
+   "isLoginRequest": true,
    "loginError": null,
    "user": Object {
      "jwt": "some jwt text",
    },
  }
`;

exports[`handle initial state 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;
